### PR TITLE
chore: Upgrade default npm to 10.2.1

### PR DESCRIPTION
**Pull request by tltv**
Fixes: #17874

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendTools.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ public class FrontendTools {
     /**
      * This is the version shipped with the default Node version.
      */
-    public static final String DEFAULT_NPM_VERSION = "10.1.0";
+    public static final String DEFAULT_NPM_VERSION = "10.2.1";
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendTools.java:73]: This change will always fail the test `installNode_NodeIsInstalledToTargetDirectory` as the npm version supplied by node 20.8.1 (and 20.9.0) is 10.1.0
Event the current node version 21.1.0 only brings in npm 10.2.0
I don't think we have a feature for updating the npm installation after node installation.

**Comment by tltv** [comment]
Need to wait that this https://github.com/npm/cli/issues/6835 gets to the next LTS node 20 release, and both node and npm default versions has to be then updated at the same time in `FrontendTools` [based on this release table](https://nodejs.org/en/download/releases).

Other options would be to update to latest node 21 but this was already decided otherwise here https://github.com/vaadin/flow/pull/17877.

**Comment by mshabarov** [comment]
Waits for https://github.com/nodejs/node/issues/50376

**Comment by Artur-** [comment]
This was upgraded in #18111

**Pull request closed without merging**